File: zmusic-plugin/src/main/java/me/zhenxin/zmusic/ZMusicBC.java
package me.zhenxin.zmusic;

import me.zhenxin.zmusic.bstats.MetricsBC;
import me.zhenxin.zmusic.command.CmdBC;
import me.zhenxin.zmusic.config.Config;
import me.zhenxin.zmusic.event.EventBC;
import me.zhenxin.zmusic.utils.CookieUtils;
import me.zhenxin.zmusic.utils.log.LogBC;
import me.zhenxin.zmusic.utils.message.MessageBC;
import me.zhenxin.zmusic.utils.mod.SendBC;
import me.zhenxin.zmusic.utils.music.Music;
import me.zhenxin.zmusic.utils.player.PlayerBC;
import me.zhenxin.zmusic.utils.runtask.RunTaskBC;
import net.md_5.bungee.api.plugin.Plugin;

public class ZMusicBC extends Plugin {
    public static Plugin plugin;

    @Override
    public void onEnable() {
        ZMusic.log = new LogBC(getProxy().getConsole());
        plugin = this;
        ZMusic.isBC = true;
        ZMusic.runTask = new RunTaskBC();
        ZMusic.message = new MessageBC();
        ZMusic.music = new Music();
        ZMusic.send = new SendBC();
        ZMusic.player = new PlayerBC();
        ZMusic.dataFolder = getDataFolder();
        if (!ZMusic.dataFolder.exists()) {
            ZMusic.dataFolder.mkdir();
        }
        Config.debug = true;
        ZMusic.thisVer = getDescription().getVersion();
        ZMusic.log.sendNormalMessage("正在加载中....");
        CookieUtils.initCookieManager();
        new MetricsBC(this, 8864);
        getProxy().registerChannel("zmusic:channel");
        getProxy().registerChannel("allmusic:channel");
        getProxy().registerChannel("AudioBuffer");
        getProxy().getPluginManager().registerCommand(this, new CmdBC());
        getProxy().getPluginManager().registerListener(this, new EventBC());
        ZMusic.loadEnd(getProxy().getConsole());
    }

    @Override
    public void onDisable() {
        ZMusic.disable();
    }

}

File: src/main/kotlin/me/zhenxin/zmusic/ZMusicBC.kt
package me.zhenxin.zmusic

import me.zhenxin.zmusic.bstats.MetricsBC
import me.zhenxin.zmusic.command.CmdBC
import me.zhenxin.zmusic.config.Lang
import me.zhenxin.zmusic.event.EventBC
import me.zhenxin.zmusic.module.logger.LoggerBC
import me.zhenxin.zmusic.module.tasker.TaskerBC
import me.zhenxin.zmusic.util.VersionCheck
import net.md_5.bungee.api.plugin.Plugin


class ZMusicBC : Plugin() {

    override fun onEnable() {
        proxy.registerChannel("zmusic:channel")
        proxy.registerChannel("allmusic:channel")
        proxy.registerChannel("AudioBuffer")
        proxy.pluginManager.registerCommand(this, CmdBC())
        proxy.pluginManager.registerListener(this, EventBC())

        MetricsBC(this, 8864)

        ZMusic.isBC = true
        ZMusic.logger = LoggerBC(proxy.console)
        ZMusic.thisVer = description.version
        ZMusic.tasker = TaskerBC()
        ZMusic.dataFolder = dataFolder
        val zLibVer = proxy.pluginManager.getPlugin("ZLib")!!.description.version
        if (VersionCheck(ZMusic.zLibVer, zLibVer).isLowerThan()) {
            ZMusic.logger.error(Lang.Loading.zLibNoOK.replace("%version%", ZMusic.zLibVer))
            ZMusic.zLibIsOK = false
        }
        ZMusic.enable()
    }
}
